language/move-lang/src/lib.rs

```diff
@@ -197,6 +210,88 @@ pub fn output_compiled_units(
     Ok(())
 }
 
+fn generate_interface_files_for_deps(
+    deps: &mut Vec<String>,
+    interface_file_output_directory_opt: Option<String>,
+) -> anyhow::Result<()> {
+    if let Some(dir) = generate_interface_files(deps, interface_file_output_directory_opt)? {
+        deps.push(dir)
+    }
+    Ok(())
+}
+
+pub fn generate_interface_files(
+    mv_file_locations: &[String],
+    interface_file_output_directory_opt: Option<String>,
+) -> anyhow::Result<Option<String>> {
+    let mv_files = find_move_filenames_by_extension(mv_file_locations, MOVE_COMPILED_EXTENSION)?
+        .into_iter()
+        .filter(|s| {
+            let p = Path::new(s);
+            let mv_file = p.is_file()
+                && p.extension()
+                    .map(|ext| ext == MOVE_COMPILED_EXTENSION)
+                    .unwrap_or(false);
+            mv_file || has_compiled_module_magic_number(s)
```

Comment:
@bob-wilson I had some issues with 'find_move_filenames_by_extension' since it always returns specific files. So I had issues with some temp files (as I could not give them custom extensions). 
I think what I have here is okay, but let me know if you have any thoughts

Reply:
The code seems unnecessarily confusing this way. As I understand it, there are 2 cases:

1. A dependency is specified as a directory. Only the files within that directory with the MOVE_COMPILED_EXTENSION will be added to the list of mv_files. In this case, there's no reason to check the extension again, and it's confusing to see that check again.
2. A filename is specified directly as a dependency. It will get through find_move_filenames_by_extension regardless of the extension match. I assume this is where you're seeing the issue with temp files. Checking the extension is not redundant in this case, but neither is it reliable. How about checking only the magic number here? It would be a bit slower because the file needs to be opened and read, but I'd be surprised if that is even noticeable.

Regardless, a comment would be helpful here to explain what's going on.

Reply:
> Only the files within that directory with the MOVE_COMPILED_EXTENSION will be added to the list of mv_files. In this case, there's no reason to check the extension again, and it's confusing to see that check again.

Ah! So I thought too! But the find files utility returns any specific files passed to it. I'll clean that up a bit